PR for SRVCOM-3402: Update the Serverless 1.35 release notes (#85643)
openshift/openshift-docs: modules/serverless-rn-1-35-0.adoc

== New features

* {ServerlessProductName} now uses Knative Serving 1.15.
* {ServerlessProductName} now uses Knative Eventing 1.15.
* {ServerlessProductName} now uses Kourier 1.15.
* {ServerlessProductName} now uses Knative (`kn`) CLI 1.15.
* {ServerlessProductName} now uses Knative for Apache Kafka 1.15.
* The `kn func` CLI plugin now uses `func` 1.16.

* Go functions using S2I builder are now available as a Generally Available (GA) feature for Linux and Mac developers, allowing them to implement and build Go functions on these platforms.

* Workflows deployed using the `dev` or `preview` profiles are now automatically configured to generate monitoring metrics for Prometheus.

* The Jobs Service supporting service can now be scaled to zero by configuring the `spec.services.jobService.podTemplate.replicas` field in the `SonataFlowPlatform` custom resource (CR).

* Workflows deployed with the `preview` and `gitops` profiles are now automatically configured to send grouped events to the Data Index, optimizing event traffic.

* A more comprehensive list of errors in the workflow definition is now provided, rather than only displaying the first detected error.

> * {ServerlessLogicProductName} is now certified for use with PostgreSQL version `15.9`.

Reviewer: Can you guys confirm this this 15.9 version is the correct: gmunozfe@redhat.com  dhanak@redhat.com

Author: Is it confirmed that version 15.9 is correct?

Reviewer 2: @domhanak can you take a look?

> * Event performance between runtimes and the Data Index has been significantly enhanced through event batching. To enable event grouping, set the `kogito.events.grouping=true` property. For further optimization, you can use the `kogito.events.grouping.binary=true` property to activate an alternative serialization algorithm that reduces the size of grouped events. Additionally, enabling `kogito.events.grouping.compress=true` applies compression to grouped events, further reducing their size at the cost of slightly increased CPU usage.

Reviewer: ```suggestion
* Event performance between {ServerlessLogicProductName} workflows and the Data Index has been significantly enhanced through event batching. To enable event grouping, set the `kogito.events.grouping=true` property. For further optimization, you can use the `kogito.events.grouping.binary=true` property to activate an alternative serialization algorithm that reduces the size of grouped events. Additionally, enabling `kogito.events.grouping.compress=true` applies compression to grouped events, further reducing their size at the cost of slightly increased CPU usage.
```

Author: Done!

* Compensation states are now invoked when a workflow is aborted.

* {ServerlessLogicProductName} now supports configuring the Knative Eventing system to produce and consume events for workflows and supporting services.

> * The secret configurations for the Broker and KafkaChannel (Apache Kafka) have been unified.

Reviewer: @kaldesai sorry, this page is global to OpenShift serverless and not only to OpenShift Serverless Logic, right?

Reviewer: f this is the case, shouldn't we somehow add a prefix {ServerlessLogicProductName} to all items related to us?

gmunozfe@redhat.com  dhanak@redhat.com zanini@redhat.com

Author: @wmedvede, That is correct! It was decided to include both OpenShift Serverless and OpenShift Serverless Logic in a single section of the release notes, rather than creating a separate section for each. Therefore, I will use the “OSL” prefix in the release notes entries wherever applicable.

// Module included in the following assemblies
//
// * about/serverless-release-notes.adoc

:_content-type: REFERENCE
[id="serverless-rn-1-35-0_{context}"]
= Red Hat {ServerlessProductName} 1.35

{ServerlessProductName} 1.35 is now available. New features, updates, and known issues that pertain to {ServerlessProductName} on {ocp-product-title} are included in the following notes:

[id="new-features-1-35-0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 1.15.
* {ServerlessProductName} now uses Knative Eventing 1.15.
* {ServerlessProductName} now uses Kourier 1.15.
* {ServerlessProductName} now uses Knative (`kn`) CLI 1.15.
* {ServerlessProductName} now uses Knative for Apache Kafka 1.15.
* The `kn func` CLI plugin now uses `func` 1.16.

* Go functions using S2I builder are now available as a Generally Available (GA) feature for Linux and Mac developers, allowing them to implement and build Go functions on these platforms.

* It is now possible to automatically discover and register `EventTypes` based on the structure of incoming events, simplifying the overall configuration and management of `EventTypes`.

> * Knative Eventing catalog is now available in OpenShift Developer Console (ODC). You can explore the catalog to discover different event types, along with their descriptions and associated metadata, making it easier to understand the system capabilities and functionalities.
> 
> * Knative Eventing now supports long-running background jobs. This feature separates resource-intensive or time-consuming tasks from the primary event processing flow, boosting application responsiveness and scalability.
> 
> * Knative Eventing catalog is now available in Backstage. Backstage is as an internal developer portal, centralizing tools and resources.

Reviewer 3: ```suggestion
* Knative Eventing catalog is now available in OpenShift Developer Console (ODC). You can explore the catalog to discover different event types, along with their descriptions and associated metadata, making it easier to understand the system capabilities and functionalities.

* Knative Eventing catalog is now available in Backstage. Backstage is as an internal developer portal, centralizing tools and resources.

* Knative Eventing now supports long-running background jobs. This feature separates resource-intensive or time-consuming tasks from the primary event processing flow, boosting application responsiveness and scalability.
```

Just a re-order of the items to "group" related ones

Author: Done!

[id="new-features-1-35-0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 1.15.
* {ServerlessProductName} now uses Knative Eventing 1.15.
* {ServerlessProductName} now uses Kourier 1.15.
* {ServerlessProductName} now uses Knative (`kn`) CLI 1.15.
* {ServerlessProductName} now uses Knative for Apache Kafka 1.15.
* The `kn func` CLI plugin now uses `func` 1.16.

* Go functions using S2I builder are now available as a Generally Available (GA) feature for Linux and Mac developers, allowing them to implement and build Go functions on these platforms.

* It is now possible to automatically discover and register `EventTypes` based on the structure of incoming events, simplifying the overall configuration and management of `EventTypes`.

* Knative Event catalog is now available in OpenShift Developer Console (ODC). You can explore the catalog to discover different event types, along with their descriptions and associated metadata, making it easier to understand the system capabilities and functionalities.

> * Knative Eventing Catalog plugin is now available in the Backstage plugin listing, and you can also install it on Red Hat Developer Hub (RHDH).

Reviewer 4: ```suggestion
* Knative Eventing Catalog plugin is now available in the Backstage plugin listing, and you can also install it on Red{nbsp}Hat Developer Hub (RHDH).
```

https://redhat-documentation.github.io/supplementary-style-guide/#non-breaking-spaces

> Use a non-breaking space ({nbsp}) between the words "Red" and "Hat". The non-breaking space prevents an automatic line break from separating the two words onto two lines.

* Knative Eventing now supports long-running background jobs. This feature separates resource-intensive or time-consuming tasks from the primary event processing flow, boosting application responsiveness and scalability.

> * Autoscaling for Knative Kafka subscriptions is now enhanced with Kubernetes Event-Driven Autoscaling (KEDA) as a Technology Preview (TP). Autoscaling with CMA/KEDA optimizes resource allocation for Kafka triggers and KafkaSource objects, boosting performance in event-driven workloads by enabling dynamic scaling of Kafka consumer resources.

Reviewer 4: ```suggestion
* Autoscaling for Knative Kafka subscriptions is now enhanced with Kubernetes Event-Driven Autoscaling (KEDA) as a Technology Preview (TP) feature. Autoscaling with CMA/KEDA optimizes resource allocation for Kafka triggers and `KafkaSource` objects, boosting performance in event-driven workloads by enabling dynamic scaling of Kafka consumer resources.
```

* {ServerlessLogicProductName} now integrates with Prometheus and Grafana to provide enhanced monitoring support.

> * {ServerlessLogicProductName} workflows deployed using the `dev` or `preview` profiles are now automatically configured to generate monitoring metrics for Prometheus.

Reviewer 4: ```suggestion
* {ServerlessLogicProductName} workflows deployed using the `dev` or `preview` profile are now automatically configured to generate monitoring metrics for Prometheus.
```

* Event performance between {ServerlessLogicProductName} workflows and the Data Index is improved through event batching. Set `kogito.events.grouping=true` to group events. For further optimization, enable `kogito.events.grouping.binary=true` to reduce the size of grouped events with an alternate serialization algorithm. To compress these events, set `kogito.events.grouping.compress=true`, which lowers event size at the cost of additional CPU usage.

* Compensation states are now invoked when a workflow is aborted.

* {ServerlessLogicProductName} now supports configuring the Knative Eventing system to produce and consume events for workflows and supporting services.

* The secret configurations for the Broker and KafkaChannel (Apache Kafka) have been unified.

[id="fixed-issues-1-35-0_{context}"]
== Fixed issues

* Previously, Horizontal Pod Autoscaler (HPA) scaled down the Activator component prematurely, causing long-running requests against a Knative Service to terminate. This issue is now fixed. The `terminationGracePeriodSeconds` value is automatically set according to the `max-revision-timeout-seconds` configuration for Knative revisions.

> * Previously, requests to a Knative Service with a slow back end could time out because the default OpenShift Route timeout was too short. You can now configure the route HAProxy timeout by specifying an environment variable in the Operator Subscription for {ServerlessProductName} as follows:

Reviewer 4: ```suggestion
* Previously, requests to a Knative Service with a slow back end could time out because the default {product-title} route timeout was too short. You can now configure the route HAProxy timeout by specifying an environment variable in the Operator `Subscription` object for {ServerlessProductName} as follows:
```
